content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,855 @@
+
+   return OUT;
+   ``` 
+   {{< note >}}
+   The code needs to be above the two `OUT` lines because you will edit `worldPosition`, which adjusts the `OUT` variables accordingly.
+   {{< /note >}}
+  
+1. Repeat steps 2-3 with the depth pass in `VegetationBending_DepthPass.azsli` and the depth pass with PS in `VegetationBending_DepthPass_WithPS.azsli`.
+
+   {{< tip >}}
+   Note that the depth pass does not output a world position. Just place the code above the `OUT.m_position` line. The depth pass with PS has some extra code in the vertex shader, but you can still place this code about the two `OUT` statements.
+   {{< /tip >}}
+
+### Set up wind bending
+Let's begin editing the code to add wind. 
+
+1. Open `VegetationBending_Common.azsli`.
+
+1. Above your `ProcessBending` function, add a function to calculate the amplitude, frequency, and phase of the wind according to the time and world position of the vertex. The wind's phase uses the `worldPosition` to mimic how wind affects nearby objects similarly, but faraway objects differently. This is because in real life, faraway objects may not be affected by the same breeze. 
+
+  Later, you'll use this function to calculate the appropriate movement of the vertex.
+
+   ```hlsl
+   float4 SetUpWindBending(float currentTime, float4 worldPosition) 
+   {
+      float2 wind = float2(MaterialSrg::m_windX, MaterialSrg::m_windY);
+      float bendingStrength = MaterialSrg::m_bendingStrength;
+      float2 amplitude = float2(wind.x * 0.4 + wind.y * 0.2, wind.y * 0.4 - wind.x * 0.2);
+      float2 frequency = float2(MaterialSrg::m_windBendingFrequency, MaterialSrg::m_windBendingFrequency * 1.125);
+      // Using the world position to modify the phase makes it so different trees near each other are at similar but not equal points in the animation, 
+      // so they appear to be reacting to the same wind but at different times as the wind moves through the vegetation.
+      float2 phase = worldPosition.xy * 0.08;
+
+      float2 bendAmount = sin(currentTime * frequency + phase) * amplitude;
+
+      float4 result;
+      result.xy = bendAmount + wind;
+      result.z = length(wind);
+      result.w = 0.3 * length(result.xy);
+      result.xyz *= bendingStrength * 0.08;
+
+      return result;
+   }
+   ```
+
+   {{< note >}}
+   This function could potentially be run once per-object on the CPU with the results updated each frame in the ObjectSrg instead of being run once per-vertex on the GPU. It's a tradeoff between doing the computation with an extra SRG compile once per frame per vegetation object versus re-doing the computation on the GPU per frame per vegetation object per vertex.
+   
+   The choice is content specific, with the redundant GPU cost increasing as vertex density increases. The choice also depends on if the GPU is the bottleneck, if the vertex shader is the bottleneck, and if the vertex shader is bandwidth bound or arithmetic logic unit (ALU) bound. 
```

Comment:
```suggestion
   By default, this function runs once per vertex on the GPU. Instead, you can potentially run it once per object on the CPU, causing the results to update each frame in the ObjectSrg. The tradeoff is between recomputing on the GPU per object, per vertex, per frame, versus computing with an extra SRG compile once per frame, per object.

   Your choice may depend on how much content you have, since the redundant GPU cost increases as vertex density increases. Your choice may also depend on whether the GPU or the vertex shader is the bottleneck, or if the vertex shader is bandwidth bound or arithmetic logic unit (ALU) bound. 
```